rfd/0067-desktop-access-file-system-sharing.md

```diff
@@ -0,0 +1,541 @@
+                       |                                    |
+                       |                                    |
+```
+
+A similar process would then take place for other operations, for example a read of a file in the shared directory named `example.txt` would look like
+
+1. server sends `IRP_MJ_CREATE` with `Path: "example.txt"`
+2. client responds with a `FileId`
+3. server sends `IRP_MJ_READ` for `FileId` for bytes 0 - 1024
+4. client executes the read and responds with the data
+5. server sends `IRP_MJ_CLOSE`
+
+## TDP File Shared Directory Extension
+
+Our RDP client lives on the Windows Desktop Service, while the directory we're sharing is exposed to us via the user's browser. This means that in order to get information to and from the shared directory,
+we must extends the TDP protocol (see the diagram in the `Introduction` section for reference).
+
+Each `* Request` and `* Response` TDP message contains a `completion_id` field, with `* Request`s being responsible for generating the `completion_id`s, and `* Response`s being responsible for
+including the correct `completion_id` to signify which `* Request` the response is intended for.
+
+#### 11 - Shared Directory Announce
+
+```
+| message type (11) | directory_id uint32 | name_length uint32 | name []byte |
+```
+
+This message announces a new directory to be shared over TDP. `directory_id` must be a unique identifier. Attempting to share multiple different directories using
+the same `directory_id` is undefined behavior.
+
+`name_length` is the length in bytes of the `name`
+
+`name` is the name of the directory (without any path prefix)
+
+#### 12 - Shared Directory Acknowledge
+
+```
+| message type (12) | err error | directory_id uint32 |
+```
+
+Acknowledges a `Shared Directory Announce` was received.
+
+`err` is a `Shared Directory Error`. `0` ("nil") means the `Shared Directory Announce` was successfully processed, `1` ("operation failed") means processing failed.
+
+`directory_id` is the `directory_id` of the top level directory being shared, as specified in the `Announce Shared Directory` message this message is acknowledging.
+
+#### 13 - Shared Directory Info Request
+
+```
+| message type (13) | completion_id uint32 | directory_id uint32 | path_length uint32 | path []byte |
+```
+
+This message is sent from the server to the client to request information about a file. The server will be expecting a `Shared Directory Info Response`.
+
+`completion_id` is generated by the server and must be returned by the client in its corresponding `Shared Directory Info Response` or `Shared Directory Error`
+response.
+
+`directory_id` is the `directory_id` of the top level directory being shared, as specified in a previous `Announce Shared Directory` message.
+
+`path_length` is the length in bytes of the `path`
+
+`path` is the unix-style relative path (from the root-level directory specified by `directory_id`) to the file or directory, excepting special path characters ".",
```

Comment:
Do we have to deal with Windows-style short paths here?

https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-2000-server/cc976806(v=technet.10)

Reply:
From my research, it seems that its no longer commonly used (i.e. [this](https://barreto.home.blog/2012/11/13/windows-server-2012-file-server-tip-disable-8-3-naming-and-strip-those-short-names-too/)). I also didn't see anything in FreeRDP that deals with it.